pkg/stolon/stolon.go

```diff
@@ -79,24 +78,28 @@ func (c Clusterdata) Master() DB {
 	return c.Dbs[c.Proxy.Spec.MasterDbUID]
 }
 
-// CheckHealthy returns an error if any of the keepers are marked as unhealthy
+// CheckHealthy returns an error if too many keepers are marked as unhealthy
+// We define healthy as having a healthy master and two other healthy keepers
 func (c Clusterdata) CheckHealthy() error {
 	if reflect.DeepEqual(c.Master(), DB{}) {
 		return errors.New("no healthy master")
 	}
 
-	errors := []string{}
+	healthy := 0
+Loop:
 	for _, db := range c.Dbs {
-		if !db.Status.Healthy {
-			errors = append(errors, db.String())
+		if db.Spec.KeeperUID == c.Master().Spec.KeeperUID {
+			continue Loop
```

Comment:
Not sure why we're using a label here? We can just continue and we'll fall through to the next for loop I believe

Reply:
Was leftover. Removed as this function has totally changed now anyway.